sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_training_client.py

```diff
@@ -122,8 +123,23 @@ def begin_train_model(self, training_files_url, use_training_labels, **kwargs):
                 :caption: Training a model with your custom forms.
         """
 
+        def callback(raw_response):
+            model = self._client._deserialize(Model, raw_response)
+            return CustomFormModel._from_generated(model)
+
         cls = kwargs.pop("cls", None)
+        continuation_token = kwargs.pop("continuation_token", None)
         polling_interval = kwargs.pop("polling_interval", POLLING_INTERVAL)
+        deserialization_callback = cls if cls else callback
+
+        if continuation_token:
+            return LROPoller.from_continuation_token(
```

Comment:
How come `begin_train_model` calls LROPoller.from_continuation_token, while the other polling methods do not call it if a continuation token is present

Reply:
This one doesn't generate as an LRO (although it is an LRO). It uses Location polling which should not be used for POST/201, from my understanding. I believe there is an issue open in autorest to handle this, but for now I have to work around it like this. The other methods do call it if continuation_token is present, but in the generated code.